File: FashionClothesAndTrends.Domain/Entities/Comment.cs
using FashionClothesAndTrends.Domain.Common;

namespace FashionClothesAndTrends.Domain.Entities;

public class Comment : BaseEntity
{
    public string Text { get; set; }

    public Guid UserId { get; set; }
    public User User { get; set; }
    
    public Guid ClothingItemId { get; set; }
    public ClothingItem ClothingItem { get; set; }
    
    public ICollection<LikeDislike> LikesDislikes { get; set; }
}

File: ClientApp/src/app/shared/models/comment.ts
import { Guid } from "guid-typescript";
import { LikeDislike } from "./like-dislike";

export interface Comment {
  id: Guid;
  text: string;
  username: string;
  userId: string;
  createdAt: Date;
  timeAgo?: string;
  likesDislikes: LikeDislike[];
}
